src/tests/cohort.integration.spec.ts

```diff
@@ -223,6 +224,23 @@ describe('Integration test for cohort saving and restoring', () => {
       });
   });
 
+  it('should ensure patient level constraint before updating counts', () => {
```

Comment:
This kind of tests is a real anti-pattern. It tests whether a particular chain of function call is being done instead of testing whether certain input results in expected output.

Reply:
it's replaced with unit tests now.